### PR TITLE
Remove rails requirement

### PR DESCRIPTION
**Pull request by vimutter**
Hi, 

requiring whole rails is not optimal since it limits applications from dropping some of rails optional gems from Gemfile. 
This gem only uses Engine and `active_support` both of which are required by devise's dependency - `railties`.

### CODE REVIEW

**Comment by olbrich** [comment]
@vimutter can you please describe which rails components you need to drop?

**Comment by vimutter** [comment]
> @vimutter can you please describe which rails components you need to drop?

Our use case was that we were not using `actioncable`, `actionmailbox`, `actiontext`, `activejob` and `activestorage`.

File: Gemfile.lock [hunk 1]
```diff
@@ -267,7 +266,7 @@ DEPENDENCIES
   pry-byebug
   pry-rescue
   rails (~> 5.0.0)
```
Comment (oniofchaos) [inline Gemfile.lock:268]: Was this file regenerated properly? This shouldn't be listed as a top level dependency if it's not in the `gemspec` anymore.

Comment (olbrich) [reply Gemfile.lock:268]: Yes, the Gemfile specifies a Rails version that we use for local development, it's not part of the gemspec

**Review by olbrich — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (olbrich) [reply Gemfile.lock]: actually, I have updated the appraisals to use railties instead of the full rails, so this should go away now.

**Review by oniofchaos — APPROVED** [review]

**Pull request merged**